### DISCUSSION TITLE
Calculate the variables and the result is incorrect.

### DISCUSSION THREAD

**Discussion by LTCTHITA** [Budibase/budibase#14164] [Help] [answered]
Hey,...

I calculate the variables At first there were 10 variables and the calculations could be done correctly, but when the variables were increased to 30, the resulting numbers increased from tens of thousands to hundreds of thousands.
![image](https://github.com/user-attachments/assets/f922ce36-01c3-44c8-b700-a41323b008fb)
![image](https://github.com/user-attachments/assets/7eec5b0f-a7b4-498b-9d24-93c499ea3508)
The code used is 

function sumVariables(...vars) {
    return vars.reduce((sum, current) => sum + (Number(current) || 0), 0);
}
let c1 = $("Scan.Fields.Var Count 1")
let c2 = $("Scan.Fields.Var Count 2")
let c3 = $("Scan.Fields.Var Count 3")
let c4 = $("Scan.Fields.Var Count 4")
let c5 = $("Scan.Fields.Var Count 5")
let c6 = $("Scan.Fields.Var Count 6")
let c7 = $("Scan.Fields.Var Count 7")
let c8 = $("Scan.Fields.Var Count 8")
let c9 = $("Scan.Fields.Var Count 9")
let c10 = $("Scan.Fields.Var Count 10")
let c11 = $("Scan.Fields.Var Count 11")
let c12 = $("Scan.Fields.Var Count 12")
let c13 = $("Scan.Fields.Var Count 13")
let c14 = $("Scan.Fields.Var Count 14")
let c15 = $("Scan.Fields.Var Count 15")
let c16 = $("Scan.Fields.Var Count 16")
let c17 = $("Scan.Fields.Var Count 17")
let c18 = $("Scan.Fields.Var Count 18")
let c19 = $("Scan.Fields.Var Count 19")
let c20 = $("Scan.Fields.Var Count 20")
let c21 = $("Scan.Fields.Var Count 21")
let c22 = $("Scan.Fields.Var Count 22")
let c23 = $("Scan.Fields.Var Count 23")
let c24 = $("Scan.Fields.Var Count 24")
let c25 = $("Scan.Fields.Var Count 25")
let c26 = $("Scan.Fields.Var Count 26")
let c27 = $("Scan.Fields.Var Count 27")
let c28 = $("Scan.Fields.Var Count 28")
let c29 = $("Scan.Fields.Var Count 29")
let c30 = $("Scan.Fields.Var Count 30")

// รวมค่าทั้งหมด
let total = sumVariables(c1, c2, c3, c4, c5, c6, c7, c8, c9, c10, c11, c12, c13, c14, c15, c16, c17, c18, c19, c20, c21, c22, c23, c24, c25, c26, c27, c28, c29, c30);

//console.log("Total:", total); // แสดงผลรวม
return total;

---------------------------------------------------

![image](https://github.com/user-attachments/assets/e2aa0ac9-a7e8-476b-9674-35410645a2f1)
I put the value in the variable
var count 1 = 17000
var count 2 = 17000
The result must be 34000.
But the result came out to be 357000.
The moment I pressed the button to calculate numbers, it first came up to 34000 and then changed to 357000 itself.

**Comment by mikesealey** [comment, marked as answer]
Hey @LTCTHITA 

I've tried to replicate this issue up to 30 fields, but I've been unable to.

The behaviour you describe - _it first came up to 34000 and then changed to 357000 itself_ - may be to do with the numbers in fields saving when the field loses focus (clicking out of the field).

My suggestion would be to totally simplify your code to try to eliminate possible issues.

Try this:

return 
$("Scan.Fields.Var Count 1") + 
$("Scan.Fields.Var Count 2") +
$("Scan.Fields.Var Count 3") +
$("Scan.Fields.Var Count 4") +
$("Scan.Fields.Var Count 5") +
$("Scan.Fields.Var Count 6") +
$("Scan.Fields.Var Count 7") +
$("Scan.Fields.Var Count 8") +
$("Scan.Fields.Var Count 9") +
$("Scan.Fields.Var Count 10") +
$("Scan.Fields.Var Count 11") +
$("Scan.Fields.Var Count 12") +
$("Scan.Fields.Var Count 13") +
$("Scan.Fields.Var Count 14") +
$("Scan.Fields.Var Count 15") +
$("Scan.Fields.Var Count 16") +
$("Scan.Fields.Var Count 17") +
$("Scan.Fields.Var Count 18") +
$("Scan.Fields.Var Count 19") +
$("Scan.Fields.Var Count 20") +
$("Scan.Fields.Var Count 21") +
$("Scan.Fields.Var Count 22") +
$("Scan.Fields.Var Count 23") +
$("Scan.Fields.Var Count 24") +
$("Scan.Fields.Var Count 25") +
$("Scan.Fields.Var Count 26") +
$("Scan.Fields.Var Count 27") +
$("Scan.Fields.Var Count 28") +
$("Scan.Fields.Var Count 29") +
$("Scan.Fields.Var Count 30")

**Comment by LTCTHITA** [comment]
Hey, @mikesealey 
 
It still hasn't.
Previewed results are correct.
![image](https://github.com/user-attachments/assets/3102efda-1c5a-490e-86c0-c40b7ae1c5e9)

But when trying to use it, it works as per the video.

https://github.com/user-attachments/assets/594e0b2a-ac1f-473f-bad2-68e6c3b0df45


The value recorded in the database is the Error value.
![image](https://github.com/user-attachments/assets/70e1b635-2a79-44e7-bbf8-593775d8cca4)
